transmission_interface/include/transmission_interface/simple_transmission.h

```diff
@@ -0,0 +1,210 @@
+
+    /**
+     * \brief Transform \e effort variables from joint to actuator space.
+     * \param[in]  jnt_data Joint-space variables.
+     * \param[out] act_data Actuator-space variables.
+     * \pre Actuator and joint effort vectors must have size 1 and point to valid data.
+     *  To call this method it is not required that all other data vectors contain valid data, and can even remain empty.
+     */
+    void joint_to_actuator() override;
+
+    std::size_t num_actuators() const override {return 1;}
+    std::size_t num_joints()    const override {return 1;}
+
+    double get_actuator_reduction() const {return reduction_;}
+    double get_joint_offset()       const {return jnt_offset_;}
+
+protected:
+    double reduction_;
+    double jnt_offset_;
+
+    hardware_interface::CommandInterface joint_position = {"", "", nullptr};
+    hardware_interface::CommandInterface joint_velocity = {"", "", nullptr};
+    hardware_interface::CommandInterface joint_effort = {"", "", nullptr};
+
+    hardware_interface::CommandInterface actuator_position = {"", "", nullptr};
+    hardware_interface::CommandInterface actuator_velocity = {"", "", nullptr};
+    hardware_interface::CommandInterface actuator_effort = {"", "", nullptr};
+  };
+
+  inline SimpleTransmission::SimpleTransmission(
+    const double reduction,
+    const double joint_offset)
+  : reduction_(reduction),
+    jnt_offset_(joint_offset)
+  {
+    if (0.0 == reduction_) {
+      throw Exception("Transmission reduction ratio cannot be zero.");
+    }
+  }
+
+  hardware_interface::CommandInterface get_by_interface(
+    const std::vector < hardware_interface::CommandInterface > & handles,
+    const std::string & interface_name)
+  {
+    const auto result =
+      std::find_if(
+      handles.cbegin(), handles.cend(), [&interface_name](
+        const auto handle) {return handle.get_interface_name() == interface_name;});
+    if (result == handles.cend()) {
+      return hardware_interface::CommandInterface("", "", nullptr);
+    }
+    return *result;
+  }
+
+  void SimpleTransmission::configure(
+    const std::vector < hardware_interface::CommandInterface > & joint_handles,
+    const std::vector < hardware_interface::CommandInterface > & actuator_handles)
+  {
+    // check that joint / act names are identical
+    // check that joint interfaces and actuator interfaces match
+
+    joint_position = get_by_interface(joint_handles, "position");
+    joint_velocity = get_by_interface(joint_handles, "velocity");
+    joint_effort = get_by_interface(joint_handles, "effort");
+
+    actuator_position = get_by_interface(actuator_handles, "position");
+    actuator_velocity = get_by_interface(actuator_handles, "velocity");
+    actuator_effort = get_by_interface(actuator_handles, "effort");
+  }
+
+  inline void SimpleTransmission::actuator_to_joint()
+  {
+    if (joint_effort && actuator_effort) {
+      joint_effort.set_value(actuator_effort.get_value() * reduction_);
```

Comment:
What if some of the members memebers are nullptr?

Reply:
We have operator bool, the above if statement check for that